File: solutions/beecrowd/3301/3301.py
import sys

for line in sys.stdin:
    huguinho, zezinho, luisinho = map(int, line.split())
    _, middle_nephew, _ = sorted([huguinho, zezinho, luisinho])
    if middle_nephew == huguinho:
        print('huguinho')
    if middle_nephew == zezinho:
        print('zezinho')
    if middle_nephew == luisinho:
        print('luisinho')

File: solutions/beecrowd/3301/3301.clj
(ns main
  (:require [clojure.string :as str]))

(defn main []
  (loop [line (read-line)]
    (when line
      (let [[huguinho zezinho luisinho] (map #(Integer/parseInt %) (str/split line #" "))
            [_ middle-nephew _] (sort [huguinho zezinho luisinho])]
        (cond
          (= middle-nephew huguinho)
          (println "huguinho")
          (= middle-nephew zezinho)
          (println "zezinho")
          (= middle-nephew luisinho)
          (println "luisinho")))
      (recur (read-line)))))

(main)
